[FIX] Table renamed crm_lead_lost_reason using rename_tables

diggy128 opened the pull request:
This fixes renaming crm_lead_lost_reason to crm_lost_reason using
openupgrade's rename_tables so that sequences are renamed as well.

Description of the issue/feature this PR addresses:

Current behavior before PR:
Migration fails at sequence with new name

Desired behavior after PR is merged:
Migration runs.



--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Commit:
[FIX] Table renamed crm_lead_lost_reason using rename_tables

This fixes renaming crm_lead_lost_reason to crm_lost_reason using
openupgrade's rename_tables so that sequences are renamed as well.

StefanRijnhart commented:
Looks like this was conditional for a reason, but maybe you can call rename tables on this one table if the table exists?

pedrobaeza commented:
This was made on purpose because the table comes from on OCA module (crm_lead_lost_reason), as stated in the comment, so this change has no sense.

StefanRijnhart commented:
@pedrobaeza the issue at hand is that the table's id sequence is not renamed. I would appreciate a proper fix.

diggy128 commented:
@pedrobaeza I guess that performing the migration in core has some meaning too, right? 
So, should it be reverted and rename the id sequence with a query?
Either case I need to fix this to proceed with my migration so I might as well contribute a valid fix.

StefanRijnhart commented:
I think you should replace the sql with something like 

if openupgrade.table_exists(cr, old):
    openupgrade.rename_tables(cr, [(old, new)])

pedrobaeza commented:
OK about the sequence renaming, but you can't do it outside the condition to check if there's the table.

Commit:
[FIX] Rename table crm_lead_lost_reason if exists using rename_tables

This fixes renaming crm_lead_lost_reason to crm_lost_reason using
openupgrade's rename_tables so that sequences are renamed as well.
Since the table is created in an OCA module, first it is checked that
the table exists

pedrobaeza reviewed (approved):
Now it's OK

pedrobaeza commented:
Thanks for the PR!
